pkg/network/ovn_kubernetes.go

```diff
@@ -1738,3 +2210,283 @@ func isV6InternalSubnetLargeEnough(conf *operv1.NetworkSpec) bool {
 	// reserve one IP for the gw, one IP for network and one for broadcasting
 	return capacity.Cmp(maxNodesNum.Add(maxNodesNum, big.NewInt(3))) != -1
 }
+
+// Determine the zone mode by looking for a known container name in multizone mode.
+func getInterConnectZoneModeForNodeDaemonSet(ds *appsv1.DaemonSet) InterConnectZoneMode {
+	for _, container := range ds.Spec.Template.Spec.Containers {
+		if container.Name == "nbdb" {
+			return zoneModeMultiZone
+		}
+	}
+	return zoneModeSingleZone
+}
+
+func isInterConnectEnabledOnMasterStatefulSet(ss *appsv1.StatefulSet) bool {
+	for _, container := range ss.Spec.Template.Spec.Containers {
+		if container.Name == "ovnkube-master" {
+			for _, c := range container.Command {
+				if strings.Contains(c, "--enable-interconnect") {
+					return true
+				}
+			}
+			break
+		}
+	}
+	return false
+}
+
+func isInterConnectEnabledOnDaemonset(ds *appsv1.DaemonSet, containerName string) bool {
+	for _, container := range ds.Spec.Template.Spec.Containers {
+		if container.Name == containerName {
+			for _, c := range container.Command {
+				if strings.Contains(c, "--enable-interconnect") {
+					return true
+				}
+			}
+			break
+		}
+	}
+	return false
+}
+
+func isInterConnectEnabledOnNodeDaemonset(ds *appsv1.DaemonSet) bool {
+	return isInterConnectEnabledOnDaemonset(ds, "ovnkube-node")
+}
+
+func isInterConnectEnabledOnMasterDaemonset(ds *appsv1.DaemonSet) bool {
+	return isInterConnectEnabledOnDaemonset(ds, "ovnkube-master")
+}
+
+// migration from single zone to multizone is about to start if:
+// - target is multizone during a simple zone migration, single zone during an upgrade from a non-interconnect version (< 4.14)
```

Comment:
It's a fallback mode in case something goes terribly wrong with multizone. We're not officially supporting it, but we'd like to have a way to fall back to single zone (so same architecture as today, but with `--enable-interconnect`) if necessary. It would be done by pushing a configmap that looks like what CNO pushes during an upgrade to 4.14, except for "temporary=false".    With the various upgrade hacks in ovnk, this is not possible right now, but after FF we plan to  evaluate how to re-enable it.

Reply:
I am just trying to track what that `duringUpgrade` parameter means and if it is used properly. I don't know if it has to do with this `simple zone migration` or if it is just used to track if we are in the first phase or the second phase of migration. The complexity is compounded when it has a default false value in method `isMigrationToMultiZoneAboutToStartOrOngoing`. I guess it is confusing if with `MigrationToMultiZone` we are referring to the full procedure or just phase2. Probably being more explicit with these things in the method names or adding doc text in these specific methods would help. Otherwise it just gives the feeling that it is not being used appropriately or that the methods have a misleading name or that I am simply not understanding haha.

Reply:
Oh you're right, the name is definitely misleading. My bad. I needed that variable because `prepareUpgradeToInterConnect` evaluates whether we're at the end of phase 1, and if so, it changes the target zone mode from singlezone to multizone, so before it patches the configmap with this new value, the target zone mode is still the old sone, so single zone.
I'll rename the variable with `assessEndOfUpgradePhase1`, which is less concise, but more self-explanatory.